### ISSUE TITLE
Commit message regex doesn't apply to PR commits?

### ISSUE THREAD

**Issue by jfbibeau** [mohamicorp/yet-another-commit-checker#245]
Hi there!
I was hoping to use YACC to validate that PR's commit messages match a given pattern, upon merging.

However the following flow doesn't appear to work:
1. Configure a regex in YACC for commit messages
2. Try to push via git -> ok, push is rejected
3. Open a branch with a test commit
4. Merge the PR via the UI, and the commit message supplied does not match the regex.

Expected:
YACC blocks the merge.

Actual result:
Merge goes through fine, and commit has invalid message.

Am I doing something wrong or this isn't a workflow that's supported? Thanks!

**Comment by george-vulov** [comment]
Hi @jfbibeau,

In step 3 that you describe, how do you create a branch with the invalid test commit? If YACC is turned on, it should have rejected the push that created that branch.

YACC is implemented as a git [pre-receive hook](https://git-scm.com/docs/githooks#pre-receive), which means it scans all commits when they're pushed to the repo. If a commit is already in the repository before YACC is turned on, merging that commit into a new branch does not trigger any checks.

Does this match what's happening in your situation? I created a [ticket](https://mohamicorp.atlassian.net/browse/YACC-252) to track the feature request for scanning merge histories. 

@malceore -- the Bitbucket WebUI issue is tracked on [YACC-110](https://mohamicorp.atlassian.net/browse/YACC-110?src=confmacro), we should have the new build with the fix published on the marketplace shortly.

**Comment by malceore** [comment]
Thanks George, can't wait!

**Comment by jfbibeau** [comment]
@george-vulov Hi George, thanks for your reply!

The commit/branch was created using the Bitbucket UI, which I guess because of YACC-110 doesn't get blocked.

But, when the PR gets merged into master, does that not create a new commit in master, which should get blocked by the pre-receive hook since it also doesn't match the regex?

**Comment by george-vulov** [comment]
@jfbibeau You're right, I verified that YACC doesn't check the merge commits created via the Bitbucket UI at all. I opened [YACC-256](https://mohamicorp.atlassian.net/browse/YACC-256) to track this bug.

To summarize issues related to this ticket, in order of priority
1. [YACC-110](https://mohamicorp.atlassian.net/browse/YACC-110) -- Edits made through the bitbucket UI are not checked.
2. [YACC-256](https://mohamicorp.atlassian.net/browse/YACC-256) -- Merge commits made through the bitbucket UI are not checked
3. [YACC-252](https://mohamicorp.atlassian.net/browse/YACC-252) -- Add the option to scan all the new commits joining a branch when a merge happens, not just the merge commit.

Thanks for reporting these issues. We hope to have a fix for YACC-110 very shortly; the other two will be in subsequent releases.

**Comment by jfbibeau** [comment]
Thank you @george-vulov !

**Comment by george-vulov** [comment]
@jfbibeau [YACC-256](https://mohamicorp.atlassian.net/browse/YACC-256) is now released (in in [version 2.0.0](https://marketplace.atlassian.com/apps/1211854/yet-another-commit-checker?hosting=server&tab=versions)), so the merge commit created by a pull request is checked just like any other merge commit.
